src/Compilers/CSharp/Portable/Parser/QuickScanner.cs

```diff
@@ -209,7 +209,7 @@ private SyntaxToken QuickScanSyntaxToken()
                 var flags = uc < charPropLength ? (CharFlags)s_charProperties[uc] : CharFlags.Complex;
 
                 state = (QuickScanState)s_stateTransitions[(int)state, (int)flags];
-                if (state == QuickScanState.Done || state == QuickScanState.Bad)
+                if (state >= QuickScanState.Done)
```

Comment:
We also need a comment here saying that this condition covers QuickScanState.Done, QuickScanState.Bad and nothing else.